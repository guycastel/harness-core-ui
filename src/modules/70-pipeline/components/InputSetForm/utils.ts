/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { SaveToGitFormInterface } from '@modules/10-common/components/SaveToGitForm/SaveToGitForm'
import type { EntityGitDetails } from 'services/pipeline-ng'

export interface GetUpdatedGitDetailsReturnType extends EntityGitDetails {
  lastObjectId?: string
  lastCommitId?: string
  baseBranch?: string
}

export const getUpdatedGitDetails = (
  isEdit: boolean,
  gitDetails: SaveToGitFormInterface | undefined,
  lastObjectId: string,
  initialGitDetails: EntityGitDetails,
  conflictCommitId?: string
): GetUpdatedGitDetailsReturnType => {
  let updatedGitDetails: GetUpdatedGitDetailsReturnType = {}
  if (gitDetails) {
    updatedGitDetails = { ...gitDetails }
    if (isEdit) {
      updatedGitDetails['lastObjectId'] = lastObjectId
      updatedGitDetails['lastCommitId'] = conflictCommitId || initialGitDetails.commitId
    }
    if (gitDetails.isNewBranch) {
      updatedGitDetails['baseBranch'] = initialGitDetails.branch
    }
  }
  return updatedGitDetails
}
