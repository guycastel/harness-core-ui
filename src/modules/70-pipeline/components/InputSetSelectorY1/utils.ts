/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { SelectedInputSetListValue } from '../InputSetSelector/SelectedInputSetList'

export const INPUT_SET_SELECTOR_PAGE_SIZE = 100

export const isInputSetItemSelected = (selected: SelectedInputSetListValue[], identifier: string): boolean => {
  return !!selected.find(item => item.value === identifier)
}
