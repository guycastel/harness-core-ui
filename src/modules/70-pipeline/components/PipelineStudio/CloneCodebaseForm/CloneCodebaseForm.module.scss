/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.gitProviderSelect {
  width: 100%;
  display: flex;
  justify-content: space-between;
  gap: 10px;

  :global {
    .Card--card {
      padding: 6px 8px !important;
      flex: 1;
    }
  }
}

.bottomMargin3 {
  margin-bottom: var(--spacing-3) !important;
}

.predefinedValue {
  font-size: 13px;
  color: var(--grey-500);
  margin-bottom: var(--spacing-medium);
}

.popover {
  width: 400px !important;
}
