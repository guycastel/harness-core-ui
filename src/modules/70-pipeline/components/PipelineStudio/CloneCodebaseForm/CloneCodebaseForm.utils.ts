/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { SelectOption } from '@harness/uicore'
import { isRuntimeInput } from '@modules/70-pipeline/utils/CIUtils'
import { ConnectorRefWidth } from '@modules/70-pipeline/utils/constants'
import { UseStringsReturn } from 'framework/strings'
import { TypesRepository } from 'services/code'

export const runtimeInputGearWidth = 58

export const getConnectorWidth = ({
  connectorWidth,
  connectorRef
}: {
  connectorWidth?: number
  connectorRef?: string
}): number | undefined => {
  if (connectorWidth) {
    // connectorRef will be undefined for use in Steps and never subtract runtimeInputGearWidth
    return !isRuntimeInput(connectorRef) ? connectorWidth : connectorWidth - runtimeInputGearWidth
  }
  return (!isRuntimeInput(connectorRef) && ConnectorRefWidth.RightBarView) || undefined
}

export const getRepositoryOptions = ({
  gitnessRepositoriesData,
  fetchingGitnessRepos,
  getString
}: {
  gitnessRepositoriesData: TypesRepository[] | null
  fetchingGitnessRepos: boolean
  getString: UseStringsReturn['getString']
}): SelectOption[] => {
  if (fetchingGitnessRepos) {
    return [{ label: getString('common.loading'), value: 'loading' }]
  }
  if (gitnessRepositoriesData?.length) {
    return gitnessRepositoriesData.map((repoData: TypesRepository) => {
      const { uid = '' } = repoData
      return {
        label: uid,
        value: uid
      }
    })
  }
  return []
}
