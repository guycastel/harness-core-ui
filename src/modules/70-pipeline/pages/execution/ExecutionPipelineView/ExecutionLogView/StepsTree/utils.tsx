/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo } from 'lodash-es'
import {
  ExecutionPipelineGroupInfo,
  ExecutionPipelineItem,
  ExecutionPipelineNode
} from '@pipeline/components/ExecutionStageDiagram/ExecutionPipelineModel'
import { ExecutionNode, InterruptEffectDTO } from 'services/pipeline-ng'
import { getExecutionNodeName } from '@pipeline/utils/execUtils'

export function getRetryInterrupts(step: ExecutionPipelineNode<ExecutionNode>): InterruptEffectDTO[] {
  return defaultTo(step?.item?.data?.interruptHistories, defaultTo(step?.group?.data?.interruptHistories, [])).filter(
    row => row.interruptType === 'RETRY'
  )
}

export function getStepDisplayName(
  step: ExecutionPipelineItem<ExecutionNode> | ExecutionPipelineGroupInfo<ExecutionNode>
): string {
  return getExecutionNodeName(step.data) || step.name
}
