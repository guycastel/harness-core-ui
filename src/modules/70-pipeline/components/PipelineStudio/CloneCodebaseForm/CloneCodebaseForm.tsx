/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { SetStateAction, Dispatch, useMemo } from 'react'
import cx from 'classnames'
import {
  TextInput,
  Text,
  MultiTypeInputType,
  Container,
  AllowedTypes,
  MultiTypeInputValue,
  FormInput
} from '@harness/uicore'
import { get } from 'lodash-es'
import { FontVariation } from '@harness/design-system'
import type { ConnectorInfoDTO } from 'services/cd-ng'
import type { UseStringsReturn } from 'framework/strings'
import { MultiTypeTextField } from '@common/components/MultiTypeText/MultiTypeText'
import {
  handleCIConnectorRefOnChange,
  ConnectionType,
  ConnectorRefInterface,
  AcceptableValue
} from '@pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm'
import { isRuntimeInput } from '@pipeline/utils/CIUtils'
import { Connectors } from '@platform/connectors/constants'
import { getCompleteConnectorUrl } from '@platform/connectors/pages/connectors/utils/ConnectorUtils'
import { FormMultiTypeConnectorField } from '@platform/connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField'
import { isMultiTypeRuntime } from '@common/utils/utils'
import type { ConfigureOptionsProps } from '@common/components/ConfigureOptions/ConfigureOptions'
import {
  GitProviderSelect,
  getGitProviderCards
} from '@modules/10-common/components/GitProviderSelect/GitProviderSelect'
import { useListRepos } from 'services/code'
import { CodebaseRuntimeInputsInterface } from '../RightBar/RightBarUtils'
import { getConnectorWidth, getRepositoryOptions, runtimeInputGearWidth } from './CloneCodebaseForm.utils'
import css from './CloneCodebaseForm.module.scss'

export default function CloneCodebaseForm({
  values,
  setFieldValue,
  connectorUrl,
  connectionType,
  setConnectionType,
  setConnectorUrl,
  connector,
  getString,
  errors,
  loading,
  accountId,
  projectIdentifier,
  orgIdentifier,
  repoIdentifier,
  branch,
  expressions,
  isReadonly,
  setCodebaseRuntimeInputs,
  codebaseRuntimeInputs,
  connectorWidth,
  connectorAndRepoNamePath,
  allowableTypes,
  codeBaseInputFieldFormName,
  onConnectorChange,
  setConnectorType,
  connectorType,
  fixRepoNameWidth,
  configureOptionsProps,
  isCodeEnabled
}: {
  values: { [key: string]: any }
  setFieldValue: (field: string, value: unknown) => void
  connectorUrl: string
  connectionType: string
  setConnectionType: Dispatch<SetStateAction<string>>
  setConnectorUrl: Dispatch<SetStateAction<string>>
  connector?: ConnectorInfoDTO
  getString: UseStringsReturn['getString']
  errors: { [key: string]: any }
  loading: boolean
  accountId: string
  projectIdentifier: string
  orgIdentifier: string
  repoIdentifier?: string
  branch?: string
  expressions: string[]
  isReadonly: boolean
  setCodebaseRuntimeInputs: Dispatch<SetStateAction<CodebaseRuntimeInputsInterface>>
  codebaseRuntimeInputs: CodebaseRuntimeInputsInterface
  connectorWidth?: number
  connectorAndRepoNamePath?: string // coming from step / input set
  allowableTypes: AllowedTypes // expression can be used for repoName
  codeBaseInputFieldFormName?: { [key: string]: string }
  onConnectorChange?: () => void // refetch onEdit connector
  setConnectorType?: Dispatch<SetStateAction<string>>
  connectorType?: string // required for getCompleteConnectorUrl on initial Add CI Stage
  fixRepoNameWidth?: boolean
  configureOptionsProps?: Partial<ConfigureOptionsProps>
  isCodeEnabled?: boolean
}): JSX.Element {
  const { data: gitnessRepositoriesData, loading: fetchingGitnessRepos } = useListRepos({
    space_ref: `${accountId}/${orgIdentifier}/${projectIdentifier}/+`
  })

  const gitProvider =
    get(values, 'provider') ?? (isCodeEnabled ? getGitProviderCards(getString)[0] : getGitProviderCards(getString)[1])
  const connectorFieldName = connectorAndRepoNamePath ? `${connectorAndRepoNamePath}.connectorRef` : 'connectorRef'
  const connectorValue = get(values, connectorFieldName)
  const repoNameFieldName = connectorAndRepoNamePath ? `${connectorAndRepoNamePath}.repoName` : 'repoName'
  const repoNameValue = get(values, repoNameFieldName)
  const repoNameWidth =
    connectorWidth && isRuntimeInput(repoNameValue) && !fixRepoNameWidth
      ? connectorWidth + runtimeInputGearWidth
      : connectorWidth
  const repositoryOptions = useMemo(
    () => getRepositoryOptions({ gitnessRepositoriesData, fetchingGitnessRepos, getString }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [gitnessRepositoriesData, fetchingGitnessRepos]
  )

  const renderConnectorField = (): JSX.Element => {
    const args = {
      name: connectorFieldName,
      width: getConnectorWidth({ connectorWidth, connectorRef: values.connectorRef }),
      error: errors?.connectorRef,
      type: [
        Connectors.GIT,
        Connectors.GITHUB,
        Connectors.GITLAB,
        Connectors.BITBUCKET,
        Connectors.AWS_CODECOMMIT,
        Connectors.AZURE_REPO
      ],
      placeholder: loading ? getString('loading') : getString('common.entityPlaceholderText'),
      accountIdentifier: accountId,
      projectIdentifier: projectIdentifier,
      orgIdentifier: orgIdentifier,
      gitScope: { repo: repoIdentifier || '', branch, getDefaultFromOtherRepo: true },
      multiTypeProps: {
        expressions,
        disabled: isReadonly,
        allowableTypes
      },
      configureOptionsProps: configureOptionsProps,
      onChange: (value: unknown, _valueType: MultiTypeInputValue, connectorRefType: MultiTypeInputType) => {
        handleCIConnectorRefOnChange({
          value: value as ConnectorRefInterface,
          connectorRefType,
          setConnectionType,
          setConnectorUrl,
          setConnectorType,
          setFieldValue,
          codeBaseInputFieldFormName,
          onConnectorChange
        })
        setCodebaseRuntimeInputs({
          ...codebaseRuntimeInputs,
          connectorRef: isRuntimeInput(value),
          repoName: isMultiTypeRuntime(connectorRefType)
        })
      }
    }
    return (
      <FormMultiTypeConnectorField
        label={getString('connector')}
        setRefValue
        tooltipProps={{ dataTooltipId: 'rightBarForm_connectorRef' }}
        {...args}
      />
    )
  }

  return (
    <>
      {isCodeEnabled && (
        <GitProviderSelect
          gitProvider={gitProvider}
          setFieldValue={setFieldValue}
          connectorFieldName={connectorFieldName}
          repoNameFieldName={repoNameFieldName}
          handleChange={(value: AcceptableValue, connectorRefType: MultiTypeInputType) => {
            handleCIConnectorRefOnChange({
              value: value as ConnectorRefInterface,
              connectorRefType,
              setConnectionType,
              setConnectorUrl,
              setConnectorType,
              setFieldValue,
              codeBaseInputFieldFormName,
              onConnectorChange
            })
          }}
        />
      )}
      {gitProvider.type !== Connectors.Harness && (
        <Container className={cx(css.bottomMargin3)}>{renderConnectorField()}</Container>
      )}
      {gitProvider.type === Connectors.Harness ? (
        <FormInput.MultiTypeInput
          name="repositoryName"
          selectItems={repositoryOptions}
          multiTypeInputProps={{
            onChange: (value: any, _typeValue, _type) => {
              if (_type === MultiTypeInputType.FIXED) {
                setFieldValue(repoNameFieldName, value.value)
              } else {
                setFieldValue(repoNameFieldName, value)
              }
            },
            selectProps: {
              items: getRepositoryOptions({ gitnessRepositoriesData, fetchingGitnessRepos, getString }),
              addClearBtn: false,
              popoverClassName: css.popover,
              allowCreatingNewItems: true
            }
          }}
          label={getString('common.selectRepository')}
        />
      ) : !isRuntimeInput(connectorValue) && connectionType === ConnectionType.Repo ? (
        <Container width={repoNameWidth}>
          <Text
            font={{ variation: FontVariation.FORM_LABEL }}
            margin={{ bottom: 'xsmall' }}
            tooltipProps={{ dataTooltipId: 'rightBarForm_repoName' }}
          >
            {getString('common.repositoryName')}
          </Text>
          <TextInput name={repoNameFieldName} value={connectorUrl} style={{ flexGrow: 1 }} disabled />
        </Container>
      ) : (
        <>
          <Container width={repoNameWidth} className={cx(css.bottomMargin3)}>
            <MultiTypeTextField
              key={`connector-runtimeinput-${codebaseRuntimeInputs.connectorRef}_${isRuntimeInput(connectorValue)}`}
              label={
                <Text
                  font={{ variation: FontVariation.FORM_LABEL }}
                  margin={{ bottom: 'xsmall' }}
                  tooltipProps={{ dataTooltipId: 'rightBarForm_repoName' }}
                >
                  {getString('common.repositoryName')}
                </Text>
              }
              name={repoNameFieldName}
              multiTextInputProps={{
                multiTextInputProps: {
                  expressions,
                  allowableTypes
                },
                disabled:
                  (!repoNameValue && loading) ||
                  isReadonly ||
                  (isRuntimeInput(connectorValue) && isRuntimeInput(repoNameValue)) // connector is a runtime input
              }}
              configureOptionsProps={configureOptionsProps}
            />
          </Container>
          {!isRuntimeInput(connectorValue) && !isRuntimeInput(repoNameValue) && connectorUrl?.length > 0 && (
            <div className={css.predefinedValue}>
              <Text lineClamp={1}>
                {getCompleteConnectorUrl({
                  partialUrl: connectorUrl,
                  repoName: repoNameValue,
                  connectorType: get(connector, 'type') || connectorType,
                  gitAuthProtocol: get(connector, 'spec.authentication.type')
                })}
              </Text>
            </div>
          )}
        </>
      )}
    </>
  )
}
