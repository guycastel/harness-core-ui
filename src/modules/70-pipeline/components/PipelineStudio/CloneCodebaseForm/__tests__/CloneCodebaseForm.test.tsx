import React from 'react'
import { fireEvent, render, waitFor, within } from '@testing-library/react'
import { AllowedTypes, MultiTypeInputType } from '@harness/uicore'
import userEvent from '@testing-library/user-event'
import { Formik } from 'formik'
import * as codeServices from 'services/code'
import { findDialogContainer, findPopoverContainer } from '@modules/10-common/utils/testUtils'
import { TestWrapper } from '@common/utils/testUtils'
import { ConnectionType } from '@modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm'
import CloneCodebaseForm from '../CloneCodebaseForm'
import { getConnectorWidth } from '../CloneCodebaseForm.utils'

const getString = (key: any): any => {
  return key
}

export const gitnessRepos = [
  {
    uid: 'nextgenui'
  },
  {
    uid: 'learning'
  }
]

jest.mock('services/code', () => ({
  useListRepos: jest.fn().mockImplementation(() => {
    return {
      data: gitnessRepos,
      refetch: jest.fn(),
      error: null,
      loading: false
    }
  })
}))

function WrapperComponent({ isCodeEnabled }: { isCodeEnabled: boolean }): JSX.Element {
  const initialValues = {
    identifier: '',
    name: '',
    description: '',
    cloneCodebase: true
  }
  return (
    <TestWrapper>
      <Formik initialValues={initialValues} onSubmit={jest.fn()}>
        {({ values, setFieldValue }) => (
          <CloneCodebaseForm
            {...{
              values,
              setFieldValue,
              connectorUrl: '',
              connectionType: ConnectionType.Account,
              setConnectionType: jest.fn(),
              setConnectorUrl: jest.fn(),
              connector: undefined,
              connectorWidth: 402,
              getString,
              errors: {},
              loading: false,
              accountId: 'accountId',
              projectIdentifier: 'projectIdentifier',
              orgIdentifier: 'orgIdentifier',
              repoIdentifier: undefined,
              branch: undefined,
              expressions: [''],
              isReadonly: false,
              setCodebaseRuntimeInputs: jest.fn(),
              codebaseRuntimeInputs: {},
              allowableTypes: [
                MultiTypeInputType.FIXED,
                MultiTypeInputType.EXPRESSION,
                MultiTypeInputType.RUNTIME
              ] as AllowedTypes,
              configureOptionsProps: { hideExecutionTimeField: true },
              isCodeEnabled
            }}
          />
        )}
      </Formik>
    </TestWrapper>
  )
}

describe('CloneCodebaseForm utils tests', () => {
  test('getConnectorWidth with some connectorWidth and connectoreRef values', () => {
    expect(getConnectorWidth({ connectorWidth: 460, connectorRef: 'gitsync2' })).toBe(460)
    expect(getConnectorWidth({ connectorWidth: 460, connectorRef: '<+input>' })).toBe(402)
    expect(getConnectorWidth({ connectorRef: '<+input>' })).toBe(undefined)
  })
})

describe('CloneCodebaseForm tests', () => {
  test('initial render with ff on', async () => {
    const { container, getByText } = render(<WrapperComponent isCodeEnabled />)
    expect(container).toMatchSnapshot()

    expect(getByText('common.gitProvider')).toBeInTheDocument()
    expect(getByText('common.harnessCodeRepo')).toBeInTheDocument()
    expect(getByText('common.thirdPartyGitProvider')).toBeInTheDocument()
    expect(getByText('common.selectRepository')).toBeInTheDocument()
  })

  test('initial render with ff off', async () => {
    const { queryByText } = render(<WrapperComponent isCodeEnabled={false} />)

    expect(queryByText('common.harnessCodeRepo')).not.toBeInTheDocument()
    expect(queryByText('common.thirdPartyGitProvider')).not.toBeInTheDocument()
    expect(queryByText('connector')).toBeInTheDocument()
    expect(queryByText('common.repositoryName')).toBeInTheDocument()
  })

  test('harness code repos still loading', async () => {
    jest.spyOn(codeServices, 'useListRepos').mockImplementationOnce(() => ({ loading: true } as any))
    const { container } = render(<WrapperComponent isCodeEnabled />)

    const selectRepo = container.querySelector('input[name="repositoryName"]')
    await userEvent.click(selectRepo as HTMLElement)
    const popover = findPopoverContainer() as HTMLElement
    const repo = within(popover).getByText('common.loading')
    expect(repo).toBeInTheDocument()
  })

  test('select repository when harness code selected', async () => {
    const { container, getByText } = render(<WrapperComponent isCodeEnabled />)

    const selectRepo = container.querySelector('input[name="repositoryName"]')
    await userEvent.click(selectRepo as HTMLElement)
    const popover = findPopoverContainer() as HTMLElement
    const repo = within(popover).getByText('nextgenui')
    await userEvent.click(repo as HTMLElement)
    expect(getByText('nextgenui')).toBeInTheDocument()
  })

  test('set repository as runtime input harness code selected', async () => {
    const { getByText, getAllByTestId } = render(<WrapperComponent isCodeEnabled />)

    expect(getAllByTestId('multi-type-button')).toHaveLength(1)
    fireEvent.click(getAllByTestId('multi-type-button')[0])

    expect(getByText('Runtime input')).toBeInTheDocument()
    fireEvent.click(getByText('Runtime input'))
    waitFor(() => {
      expect(getByText('<+input>')).toBeInTheDocument()
    })
  })

  test('switch to third-party git provider and open connector modal', async () => {
    const { container, getByText } = render(<WrapperComponent isCodeEnabled />)

    fireEvent.click(getByText('common.thirdPartyGitProvider'))
    expect(getByText('connector')).toBeInTheDocument()
    expect(getByText('common.repositoryName')).toBeInTheDocument()

    const selectConnector = container.querySelector('[data-testid="cr-field-connectorRef"]')!
    fireEvent.click(selectConnector)
    const dialog = findDialogContainer() as HTMLElement
    expect(dialog).toBeInTheDocument()
  })

  test('switch to third-party git provider and type repo name', async () => {
    const { container, getByText } = render(<WrapperComponent isCodeEnabled />)

    fireEvent.click(getByText('common.thirdPartyGitProvider'))
    fireEvent.change(container.querySelector('input[name="repoName"]')!, { target: { value: 'repo1' } })
    waitFor(() => {
      expect(getByText('repo1')).toBeInTheDocument()
    })
  })
})
