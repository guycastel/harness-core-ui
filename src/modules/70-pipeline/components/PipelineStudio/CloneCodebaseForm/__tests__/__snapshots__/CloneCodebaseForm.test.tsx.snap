// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloneCodebaseForm tests initial render with ff on 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--padding-bottom-medium"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-form-label StyledProps--margin-bottom-small"
    >
      common.gitProvider
    </p>
    <div
      class="gitProviderSelect"
    >
      <div
        class="bp3-card bp3-interactive bp3-elevation-0 Card--card Card--selected Card--interactive"
        data-index="0"
        tabindex="0"
      >
        <span
          class="Card--corner"
        >
          <span
            class="Card--badge"
          />
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-white"
            data-icon="main-tick"
          >
            <svg
              height="10"
              viewBox="0 0 11 8"
              width="10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.219.894c.356-.38.982-.413 1.398-.072.391.32.457.857.167 1.236l-.06.07-5.138 5.486c-.362.386-.993.41-1.403.068l-.07-.064-3.2-3.214A.88.88 0 01.95 3.098a1.045 1.045 0 011.328-.003l.074.067 2.443 2.454L9.22.894z"
                fill="currentColor"
              />
            </svg>
          </span>
        </span>
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-start"
        >
          <span
            class="bp3-icon StyledProps--main"
            data-icon="code"
          >
            <svg
              fill="none"
              height="22"
              viewBox="0 0 40 40"
              width="22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="code_svg__a"
                x1="10.644"
                x2="36.25"
                y1="3.75"
                y2="36.25"
              >
                <stop
                  offset="0"
                  stop-color="#f96459"
                />
                <stop
                  offset="1"
                  stop-color="#ea473b"
                />
              </lineargradient>
              <path
                clip-rule="evenodd"
                d="M20 36.25c8.975 0 16.25-7.275 16.25-16.25S28.975 3.75 20 3.75c-2.575 0-5.01.6-7.175 1.666l4.409 4.164a3.04 3.04 0 011.732-.541c1.713 0 3.102 1.42 3.102 3.173 0 .566-.145 1.098-.399 1.558l4.262 4.026c.398-.189.84-.295 1.308-.295 1.713 0 3.102 1.421 3.102 3.174s-1.389 3.173-3.102 3.173c-1.714 0-3.102-1.42-3.102-3.173 0-.501.113-.975.315-1.397l-4.352-4.11a3.031 3.031 0 01-.772.197l-2.306 11.1a3.189 3.189 0 011.427 2.672c0 1.753-1.389 3.174-3.102 3.174-1.714 0-3.103-1.421-3.103-3.174 0-1.635 1.21-2.982 2.764-3.155l2.303-11.085a3.188 3.188 0 01-1.447-2.685c0-.32.046-.63.133-.921l-5.06-4.78C6.6 9.429 3.75 14.38 3.75 20c0 8.975 7.275 16.25 16.25 16.25z"
                fill="url(#code_svg__a)"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <div
            class="StyledProps--font StyledProps--main StyledProps--padding-left-xsmall"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-primary7"
            >
              common.harnessCodeRepo
            </p>
          </div>
        </div>
      </div>
      <div
        class="bp3-card bp3-interactive bp3-elevation-0 Card--card Card--interactive"
        data-index="1"
        tabindex="0"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-start"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
            data-icon="service-github"
          >
            <svg
              height="20"
              viewBox="0 0 14 14"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.6 6.4l-6-6C7.3.1 6.7.1 6.4.4L5.1 1.6l1.6 1.6c.4-.1.8 0 1.1.3s.4.7.2 1.1l1.5 1.5c.4-.1.8 0 1.1.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0c-.3-.3-.3-.7-.2-1.1L7.5 5.3V9c.1 0 .2.1.3.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0-.4-1.1 0-1.5c.1-.1.2-.2.3-.2V5.2c-.1-.1-.2-.1-.3-.2-.3-.4-.4-.8-.2-1.2L4.5 2.3.4 6.4c-.3.3-.3.9 0 1.2l6 6c.3.3.9.3 1.2 0l6-6c.3-.3.3-.9 0-1.2z"
                fill="#f42534"
              />
            </svg>
          </span>
          <div
            class="StyledProps--font StyledProps--main StyledProps--padding-left-xsmall"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey800"
            >
              common.thirdPartyGitProvider
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="bp3-form-group"
  >
    <label
      class="bp3-label"
      for="repositoryName"
    >
      common.selectRepository
       
      <span
        class="bp3-text-muted"
      />
    </label>
    <div
      class="bp3-form-content"
    >
      <div
        class="Layout--horizontal StyledProps--main MultiTypeInput--main"
      >
        <div
          class="bp3-popover-wrapper MultiTypeInput--select MultiTypeInput--fixedValueInput Select--main bp3-fill"
        >
          <div
            class="bp3-popover-target"
          >
            <div
              class="bp3-input-group"
            >
              <input
                autocomplete="off"
                class="bp3-input"
                name="repositoryName"
                placeholder="Search..."
                style="padding-right: 0px;"
                type="text"
                value=""
              />
              <span
                class="bp3-input-action"
              >
                <span
                  class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                  icon="chevron-down"
                >
                  <svg
                    data-icon="chevron-down"
                    height="14"
                    viewBox="0 0 16 16"
                    width="14"
                  >
                    <desc>
                      chevron-down
                    </desc>
                    <path
                      d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
        <span
          class="bp3-popover-wrapper MultiTypeInput--wrapper"
        >
          <span
            class="bp3-popover-target"
          >
            <button
              class="MultiTypeInput--btn MultiTypeInput--FIXED"
              data-testid="multi-type-button"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="fixed-input"
              >
                <svg
                  height="12"
                  viewBox="0 0 9 9"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
